test/test_reductions.py

```diff
@@ -1501,6 +1501,11 @@ def test_prod_bool(self, device):
             result = torch.prod(torch.tensor(val, device=device)).item()
             expect = np.prod(np.array(val))
             self.assertEqual(result, expect)
+        # https://github.com/pytorch/pytorch/issues/127866
+        val = [False] * 256
```

Comment:
why not adding this `val` to `vals` and tested together?

Reply:
Thanks for advice, updated.